docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-enable-cluster-label-verification/_index.md

```diff
@@ -0,0 +1,96 @@
+
+Cluster label will prevent such situation by only allowing communication for components that has same cluster label. Once that is enabled, before memberlist try to communicate with other components, it will verify whether that new components are having same cluster label and only allowing memberlist to communicate if the parties are having the same cluster label.
+
+## Migration
+
+### 1. Disable memberlist cluster label verification
+
+Cluster label verification flag is enabled by default. However, cluster label default value is an empty string. If different systems that using memberlist and not setting the cluster label, they can talk with each other. To disable cluster label verification flag, set the following structured config.
+
+```
+mimir
+  structuredConfig:
+    memberlist:
+      cluster_label_verification_disabled: true
+```
+
+Rollout the installation to apply the configuration changes by running `helm upgrade mimir-distributed -f values.yaml`.
+
+### 2. Set cluster label to all Mimir components
+
+Set cluster label to all Mimir components by setting the following configuration. Once the configuration is applied, all Mimir components will only communicate via memberlist if the other component is having the same cluster label.
+
+```
+mimir
+  structuredConfig:
+    memberlist:
+      cluster_label_verification_disabled: true
+      cluster_label: "$helm-release-name"
+```
+
+Apply the configuration changes by running `helm upgrade mimir-distributed -f values.yaml`.
+
+### 3. Enable memberlist cluster label verification
+
+Set `memberlist.cluster_label_verification_disabled` to false, to re-enable memberlist cluster label verification.
+
+```
+mimir
+  structuredConfig:
+    memberlist:
+      cluster_label_verification_disabled: false
+      cluster_label: "$helm-release-name"
+```
+
+Apply the configuration changes by running `helm upgrade mimir-distributed -f values.yaml`.
+
+## Verifying The Migration
+
+Once the rollout has been done run the following helm command to verify if we have successfully migrated the Mimir memberlist to check the cluster label.
+
+```
+helm --kube-context=[your-k8s-context] --namespace=[your-mimir-namespace] get values [your-mimir-release-name]
+```
+
+You should see the following config the in the values as what we have set above.
+
+```
+mimir
+  structuredConfig:
+    memberlist:
+      cluster_label_verification_disabled: false
+      cluster_label: "$helm-release-name"
+```
+
+{{% docs/reference %}}
+[memberlist]: "/ -> /docs/mimir/<MIMIR_DOCS_VERSION>/references/architecture/memberlist-and-the-gossip-protocol"
+[hash ring]: "/ -> /docs/mimir/<MIMIR_DOCS_VERSION>/references/architecture/hash-ring"
+{{% /docs/reference %}}
```

Comment:
I don't think there is any need to use `docs/reference` here because the links are the same and this content isn't reused via Hugo mounts as far as I know. You only need to use that shortcode for [content that is used in multiple website projects](https://grafana.com/docs/writers-toolkit/write/links/#link-from-source-content-thats-used-in-multiple-projects) like being both in Grafana and Grafana Cloud documentation.

Instead, you can just use URLs with `<MIMIR_DOCS_VERSION>` for version substitution.